Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -780,6 +780,49 @@ theorem edges_injective {u v : V} : Function.Injective (Walk.edges : G.Walk u v
 theorem darts_injective {u v : V} : Function.Injective (Walk.darts : G.Walk u v → List G.Dart) :=
   edges_injective.of_comp
 
+/-- The `Set` of edges of a walk. -/
+def edgeSet {u v : V} (p : G.Walk u v) : Set (Sym2 V) := {e | e ∈ p.edges}
+
+@[simp]
+lemma mem_edgeSet {u v : V} {p : G.Walk u v} {e : Sym2 V} : e ∈ p.edgeSet ↔ e ∈ p.edges := Iff.rfl
+
+@[simp]
+lemma edgeSet_nil (u : V) : (nil : G.Walk u u).edgeSet = ∅ := by
```

Comment:
We could (after Nil is defined) also have that `p.edgeSet = empty iff p.IsNil`. There's a few other lemmas I can think of for the new definition, basically just describing the interaction with other defs in this file (I have a rough mental heuristic that there should be about 1/2 n^2 lemmas for n definitions, describing how each def relates to each other (within reason, of course!)). For example, `(p.drop _).edgeSet` is a subset of `p.edgeSet`, the vertices mentioned are equal to the vertices in `p.support`, etc. I don't have a strong preference if you do that here, in another PR, or just leave it as TODO (in fact I have a weak preference for it to be in another PR)

Reply:
Ok, I'll do another PR after this one.